query/physicalplan/aggregate.go

```diff
@@ -103,21 +107,26 @@ func chooseAggregationFunction(
 	}
 }
 
+// AggregateColumn groups together some lower level primitives to aggregate.
+type AggregateColumn struct {
+	expr       logicalplan.Expr
+	resultName string
+	function   AggregationFunction
+	arrays     []builder.ColumnBuilder // TODO: These can actually live outside this struct and be shared. Only at the very end will they be read by each column and then aggregated separately.
+}
+
 type AggregationFunction interface {
 	Aggregate(pool memory.Allocator, arrs []arrow.Array) (arrow.Array, error)
 }
 
 type HashAggregate struct {
 	pool                  memory.Allocator
 	tracer                trace.Tracer
-	resultColumnName      string
+	columns               []AggregateColumn
```

Comment:
nit: I think it'd be nice to make the naming a bit less generic. Are these the values we're aggregating? Maybe `columnsToAggregate`

Reply:
Yes, agreed. The naming isn't perfect. It's not even about columns really. 
Given a query like `select sum(value), count(value), max(value) group by labels.label2` it's all the same `value` column, but different aggregations. So I'll probably call it `[]Aggregation`.